main/lsp/requests/go_to_implementation.cc

```diff
@@ -0,0 +1,113 @@
+#include "main/lsp/requests/go_to_implementation.h"
+#include "core/lsp/QueryResponse.h"
+#include "main/lsp/json_types.h"
+#include "main/lsp/lsp.h"
+
+using namespace std;
+
+namespace sorbet::realmain::lsp {
+
+vector<core::SymbolRef> findChildren(const core::GlobalState &gs, core::SymbolRef ancestorClass) {
+    vector<core::SymbolRef> results;
+    auto used = gs.classAndModulesUsed();
+    for (u4 idx = 1; idx < used; idx++) {
+        core::SymbolRef ref(gs, core::SymbolRef::Kind::ClassOrModule, idx);
+        if (ref.data(gs)->derivesFrom(gs, ancestorClass)) {
+            results.push_back(ref);
+        }
+    }
+    return results;
+}
+
+variant<vector<core::Loc>, unique_ptr<ResponseError>> findMethodImplementations(const core::GlobalState &gs,
```

Comment:
Feel free to make this a simple `struct` and define it in the anonymous namespace. I think that'll be easier to code against than variants, which are a bit awkward, and will be faster to compile.

Reply:
I've replaced `variant` with custom struct. Can you explain why it'll be faster? I don't really understand c++, but would be glad to learn